### PR TITLE
Added support for jpeg and gif.

### PR DESCRIPTION
**Pull request by tperale**
I only tested those for now.

### CODE REVIEW

File: src/display.c [hunk 1]
```diff
@@ -149,6 +149,57 @@ static uint32_t draw_text(cairo_t *cr, const char *text, offset_t offset, color_
...
+      /* Finding the best proportion to fit the picture. */
+      new_format.width = prop * width;
+      new_format.height = prop * height;
+
+      debug("Resizing the image to %ix%i (prop = %f)\n", new_format.width, new_format.height, prop);
+  }
+  return new_format;
+}
+
+/* @brief Draw a png at the given offset.
```
Comment (0ion9) [inline src/display.c:177]: This comment seems to be out of date. 'Draw an image' seems more appropriate.

File: Makefile [hunk 2]
```diff
@@ -7,7 +7,7 @@ CFLAGS+=-I$(INCDIR)
 SRCS=$(wildcard $(SRCDIR)/*.c)
 OBJS=$(patsubst $(SRCDIR)/%.c,$(OBJDIR)/%.o,$(SRCS))
 
-CFLAGS+=-O2 -Wall -std=c99
+CFLAGS+=-O2 -Wall -std=c99 `pkg-config --cflags --libs gtk+-3.0`
```
Comment (emgram769) [inline Makefile:10]: this adds a pretty big dependency, no?

Comment (tperale) [inline Makefile:10]: Yes sure, but I only use that part ( https://developer.gnome.org/gdk-pixbuf/2.30/ ). Feel free to revert my push if it was not the aim of the project but it also fix some errors like this https://github.com/emgram769/lighthouse/issues/33 .

Comment (0ion9) [inline Makefile:10]: It seems incorrect anyway. Don't you really want `pkg-config --cflags --libs gdk-pixbuf-2.0`? gdk-pixbuf is a separate package for exactly this reason AFAIK.

(I'm not sure exactly what is going on with GDKPixbuf in relation to gtk3, but I'm pretty sure we don't need to even try to link gtk3 to get the library we actually want here)

Comment (tperale) [inline Makefile:10]: Yeah just figured out I can just use the package name gdk... Will change it in my next commit, thank you !

**Comment by emgram769** [comment]
`pkg-config --cflags --libs gdk-pixbuf-2.0` doesn't work right off the bat because there are some functions that use gdk/gdk.h in the code. in the meantime, I pushed a change to have the makefile check for the library first before compiling.

ideally lighthouse would be as lightweight as possible. adding library dependencies is kind of messy, imo.  gdk-pixbuf alone doesn't seem terrible, thought

**Pull request merged**